### ISSUE TITLE
Validate function size before running the solver

### ISSUE THREAD

**Issue by jorisv** [Simple-Robotics/aligator#117]
Solver must throw an exception if some cost, constraints or dynamics mismatch the problem size.

**Comment by ManifoldFR** [comment]
I agree. So far all we have is validation for some cost or dynamics' models constructors, but nothing checks everything yet.

This will have to wait for the upcoming rewrite though

**Comment by ManifoldFR** [comment]
Related: #208
